### PR TITLE
Feature/otra-formacion-real-endpoint#121

### PR DESCRIPTION
**Pull request by Ric-De**
This issue have the goal of create an endpoint to return real data from the database as output.

First we made migrations, model, factories and seeders to provide data and a database feeded with fakers. Second we build the controller and the route. Finally, the old test archive as rebuilded to respond to the new controller connected to the database.
The information provided in the database is correctly showed through Swagger. Also, standard tests were created to verify the controller and passed in the test.

### CODE REVIEW

File: app/Http/Controllers/api/AdditionalTrainingListController.php [hunk 1]
```diff
@@ -1,32 +1,37 @@
...
-            }
-        
-        }
-    
-        return response()->json([
-            'additionalTraining' => $selectedProjects
-        ],200);
+    public function __invoke($uuid){
+
+        $student = Student::where('id', $uuid)->with('resume')->firstOrFail();
```
Comment (jordimorillo) [inline app/Http/Controllers/api/AdditionalTrainingListController.php:13]: Extrae de aqui el service

**Review by jordimorillo — APPROVED** [review]

**Comment by jordimorillo** [comment]
@Ric-De @AlejandroNPereyra os comparto esta rama donde he desacoplado el servicio para que os sirva de referencia:

https://github.com/IT-Academy-BCN/ita-profiles-backend/tree/feature/jordi-otra-formacion-real-endpoint%23121

File: app/Http/Controllers/api/AdditionalTrainingListController.php [hunk 2]
```diff
@@ -1,32 +1,37 @@
...
+                    'course_beggining_year' => $additionalTraining->course_beggining_year,
+                    'course_ending_year' => $additionalTraining->course_ending_year,
+                    'duration_hrs' => $additionalTraining->duration_hrs,
+                ];
+            })
+        ];
+
+        return response()->json($additionalTrainingDetail);
     }
+
```
Comment (coxen34) [inline app/Http/Controllers/api/AdditionalTrainingListController.php:36]: Array vacío
**I already told you this but in swagger it returns an empty array...**

File: app/Annotations/OpenApi/controllersAnnotations/additionalTrainingAnnotations/AnnotationsAdditionalTraining.php [hunk 3]
```diff
@@ -5,9 +5,23 @@ class AnnotationsAdditionalTraining
 {
 /**
  * @OA\Get(
- *     path="/additional-training/list",
+ *     path="/students/{uuid}/additionaltraining",
```
Comment (jordimorillo) [inline app/Annotations/OpenApi/controllersAnnotations/additionalTrainingAnnotations/AnnotationsAdditionalTraining.php:8]: If you change the path, remember to tell the frontend people that are applying this endpoints at this moment

**Review by jordimorillo — CHANGES_REQUESTED** [review]

File: database/migrations/2024_04_09_131241_add_additional_trainings_to_resumes_table.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('resumes', function (Blueprint $table) {
+            $table->json('additional_trainings_ids')->default(new Expression('(JSON_ARRAY())'))->before('created_at');
```
Comment (coxen34) [inline database/migrations/2024_04_09_131241_add_additional_trainings_to_resumes_table.php:16]: :thumbsup:

File: app/Http/Controllers/api/AdditionalTrainingListController.php (hunk 2 above)
Comment (coxen34) [reply app/Http/Controllers/api/AdditionalTrainingListController.php:36]: Solved, the problem was mine for not doing migrate:fresh --seed

File: app/Annotations/OpenApi/controllersAnnotations/additionalTrainingAnnotations/AnnotationsAdditionalTraining.php (hunk 3 above)
Comment (AlejandroNPereyra) [reply app/Annotations/OpenApi/controllersAnnotations/additionalTrainingAnnotations/AnnotationsAdditionalTraining.php:9]: ok, next task, we'll have it in mind! Thanks!

File: app/Http/Controllers/api/AdditionalTrainingListController.php (hunk 2 above)
Comment (Ric-De) [reply app/Http/Controllers/api/AdditionalTrainingListController.php:18]: Goog. Problem solved.

**Review by jordimorillo — APPROVED** [review]

**Pull request merged**